packages/nextcloud/lib/src/api/webdav/utils/webdav_response_helpers.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:nextcloud/src/api/webdav/models/models.dart';
+
+/// Helpers for WebDavResponse
+extension WebDavResponseHelpers on WebDavResponse {
+  /// All available props.
+  WebDavProp get props => propstats.singleWhere((propstat) => propstat.status.contains('200 OK')).prop;
+
+  /// The path of the resource.
+  PathUri get path {
+    final href = PathUri.parse(Uri.decodeFull(this.href!));
```

Comment:
There can be responses without a href which are not valid. Consumers need to filter these out (we already did in `toWebDavFiles()`) and then we need to prevent null hrefs here.

Reply:
But this means that unfiltered responses will throw here.
I'd rather return a `null` when there is no `href`.

Then the filtering can be done based on the `path`.